nsxt/resource_nsxt_transport_node.go

```diff
@@ -0,0 +1,1959 @@
+		Optional:    true,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"assigned_by_dhcp": {
+					Type:     schema.TypeList,
+					MaxItems: 1,
+					Optional: true,
+					//TODO: as ExactlyOneOf cannot be used, use some alternate validation
+					//ExactlyOneOf: []string{
+					//	varPath + ".0.static_ip_list_spec",
+					//	varPath + ".0.static_ip_mac_list_spec",
+					//	varPath + ".0.static_ip_pool_spec",
+					//},
+					Elem: &schema.Resource{
+						Schema: map[string]*schema.Schema{
+							"enabled": {
+								Type:        schema.TypeBool,
+								Required:    true,
+								Description: "Enables DHCP assignment. Should be set to true",
+							},
+						},
+					},
+				},
+				"static_ip_list_spec": {
+					Type:     schema.TypeList,
+					MaxItems: 1,
+					Optional: true,
+					Elem: &schema.Resource{
+						Schema: map[string]*schema.Schema{
+							"default_gateway": {
+								Type:         schema.TypeString,
+								Required:     true,
+								Description:  "Gateway IP",
+								ValidateFunc: validateSingleIP(),
+							},
+							"ip_list": {
+								Type:        schema.TypeList,
+								Description: "List of IPs for transport node host switch virtual tunnel endpoints",
+								MinItems:    1,
+								Required:    true,
+								Elem: &schema.Schema{
+									Type:         schema.TypeString,
+									ValidateFunc: validateSingleIP(),
+								},
+							},
+							"subnet_mask": {
+								Type:         schema.TypeString,
+								Required:     true,
+								Description:  "Subnet mask",
+								ValidateFunc: validateSingleIP(),
+							},
+						},
+					},
+				},
+				"static_ip_mac_list_spec": {
+					Type:     schema.TypeList,
+					MaxItems: 1,
+					Optional: true,
+					Elem: &schema.Resource{
+						Schema: map[string]*schema.Schema{
+							"default_gateway": {
+								Type:         schema.TypeString,
+								Required:     true,
+								Description:  "Gateway IP",
+								ValidateFunc: validateSingleIP(),
+							},
+							"ip_mac_list": {
+								Type:        schema.TypeList,
+								Description: "List of IPs and MACs for transport node host switch virtual tunnel endpoints",
+								MinItems:    1,
+								Required:    true,
+								Elem: &schema.Resource{
+									Schema: map[string]*schema.Schema{
+										"ip": {
+											Type:         schema.TypeString,
+											Required:     true,
+											Description:  "IP address",
+											ValidateFunc: validateSingleIP(),
+										},
+										"mac": {
+											Type:         schema.TypeString,
+											Optional:     true,
+											Description:  "MAC address",
+											ValidateFunc: validation.IsMACAddress,
+										},
+									},
+								},
+							},
+							"subnet_mask": {
+								Type:         schema.TypeString,
+								Required:     true,
+								Description:  "Subnet mask",
+								ValidateFunc: validateSingleIP(),
+							},
+						},
+					},
+				},
+				"static_ip_pool_spec": {
```

Comment:
any reason for this to be a list?

Reply:
I've defined `assigned_by_dhcp`, `static_ip_pool_spec` as lists as ExactlyOneOf demanded to have lists of MaxItems=1. Yet also the parent objects should be so and this is not the case for `transport_node_profile_sub_config`. I'm looking for some solution for this.